docs/adr/0023-paypal-support.md

```diff
@@ -0,0 +1,102 @@
+
+Pros:
+
+- Stripe continues to serve as an accurate source of truth regarding a customers subscription.
+- All the logic that determines active subscriptions and entitlements using Stripe can be used as-is.
+- Multiple product support works as-is.
+- Support representatives can see accurate payment history in Stripe for customers paying via PayPal.
+- Support can credit a PayPal customer, just as they can for a credit card customer.
+- Can extend in a similar manner to track a correlating IAP for a user.
+- Can handle switching payment method from credit card to PayPal reference ID.
+
+Cons:
+
+- Additional metadata in Stripe can start pushing us closer to metadata limits on objects.
+- Sensitive processing logic that runs PayPal transactions which we need to ensure work correctly.
+- We still could end up storing PayPal reference ID's in a separate db table to ensure we track
+  prior PayPal billing.
+
+### PayPal Subscriptions linked/tracked by FxA
+
+In this approach, we would create products/plans in PayPal, and link them back to the Stripe equivilant by updating Stripe metadata on the products/plans. FxA would also gain an additional database table that maps PayPal subscription id's to the Fx account. All the logic in FxA that looks up a users subscriptions would be updated to check the PayPal linking table to determine what subscriptions the user has active.
+
+PayPal would handle running the charges on a recurring basis per the PayPal plan that is setup. FxA would periodically check the subscription's via PayPal API to ensure they're still actively being paid (PayPal has fairly lenient handling on whether payments occur) on the schedule we want.
+
+Pros:
+
+- PayPal is responsible for charging the customer and managing the subscription lifecycle.
+- Can extend in a similar manner to track a correlating IAP for a user.
+
+Cons:
+
+- All the logic that determines active subscriptions and entitlements needs additional
+  logic/abstractions to query the database table and map active PayPal subscriptions to the
+  Stripe subscription.
+- Ability to easily cancel/re-subscribe on management page is lost, as canceled PayPal subscriptions
+  cannot be un-canceled.
+- To reduce complexity, anyone paying via PayPal would be locked into using PayPal for later
+  subscriptions and users paying via Stripe would not be presented with any PayPal payment options.
+  This could frustrate users that want to switch, as they'd have to cancel, wait till the
+  subscription expires, and then subscribe again with PayPal.
+- Stripe subscription features can't be used, such as:
+  - Crediting a user for a month
+  - Coupons
+  - Consolidating multiple plans into a single subscription for aligned billing
+- Every additional subscription requires a full PayPal authorization flow.
+- PayPal occasionally misses delivery of webhooks, which requires us to build our own script
```

Comment:
Just curious: how do we know that PayPal isn't reliable with their webhooks?

Reply:
I unfortunately have personal experience running a separate system where this has occurred multiple times. The developer dashboard for PayPal is pretty lacking in this regard, as they don't seem to retry delivery either. Typical advice is to use PayPal API's to verify actions have occurred just to be sure.